src/pyspiffe/svid/x509_svid.py

```diff
@@ -1,34 +1,472 @@
+    certs_chain_path: str,
+    encoding: serialization.Encoding,
+    x509_svid: 'X509Svid',
+) -> None:
+    try:
+        with open(certs_chain_path, 'wb') as chain_file:
+            os.chmod(chain_file.name, _CERTS_FILE_MODE)
+            for cert in x509_svid.cert_chain():
+                _write_cert_to_file(cert, chain_file, encoding)
+    except Exception as err:
+        raise StoreCertificateError(
+            'Error opening certs chain file: {}'.format(str(err))
+        )
+
+
+def _write_cert_to_file(
+    cert: Certificate, chain_file: BinaryIO, encoding: serialization.Encoding
+) -> None:
+    try:
+        cert_bytes = _extract_cert_bytes(cert, encoding)
+        chain_file.write(cert_bytes)
+    except Exception as err:
+        raise StoreCertificateError(
+            'Error writing certs chain to file: {}'.format(str(err))
+        )
+
+
+def _extract_cert_bytes(cert: Certificate, encoding: serialization.Encoding) -> bytes:
+    try:
+        cert_bytes = cert.public_bytes(encoding)
+    except Exception as err:
+        raise X509SvidError(
+            'Could not get certs chain key bytes from object: {}'.format(str(err))
+        )
+
+    return cert_bytes
+
+
+def _parse_der_private_key(private_key_bytes: bytes) -> _PRIVATE_KEY_TYPES:
+    try:
+        return load_der_private_key(private_key_bytes, None, None)
+    except Exception as err:
+        raise ParsePrivateKeyError(normalize_exception_message(str(err)))
+
+
+def _parse_pem_private_key(private_key_bytes: bytes) -> _PRIVATE_KEY_TYPES:
+    try:
+        return load_pem_private_key(private_key_bytes, None, None)
+    except Exception as err:
+        raise ParsePrivateKeyError(normalize_exception_message(str(err)))
+
+
+def _parse_der_certificates(chain_der_bytes: bytes) -> List[Certificate]:
+    result = []
+    try:
+        leaf = x509.load_der_x509_certificate(chain_der_bytes, default_backend())
+        result.append(leaf)
+        _, remaining_data = decode(chain_der_bytes)
+        while len(remaining_data) > 0:
+            cert = x509.load_der_x509_certificate(remaining_data, default_backend())
+            result.append(cert)
+            _, remaining_data = decode(remaining_data)
+    except Exception as err:
+        raise ParseCertificateError(normalize_exception_message(str(err)))
+
+    if len(result) < 1:
+        raise ParseCertificateError(_UNABLE_TO_LOAD_CERTIFICATE)
+
+    return result
+
+
+def _parse_pem_certificates(chain_pem_bytes: bytes) -> List[Certificate]:
+    result = []
+    parsed_certs = pem.parse(chain_pem_bytes)
+    for cert in parsed_certs:
+        try:
+            x509_cert = x509.load_pem_x509_certificate(
+                cert.as_bytes(), default_backend()
+            )
+            result.append(x509_cert)
+        except Exception:
+            raise ParseCertificateError(_UNABLE_TO_LOAD_CERTIFICATE)
+
+    if len(result) < 1:
+        raise ParseCertificateError(_UNABLE_TO_LOAD_CERTIFICATE)
+    return result
```

Comment:
If I got this correct, if there is a cert that cannot be parse, then it will raise an ParseCertificateError. On the other hand, if there is no cert at all, it will raise an ParseCertificateError too.

Does it makes sense to change this a little bit like this:
```
def _parse_pem_certificates(chain_pem_bytes: bytes) -> List[Certificate]:
    parsed_certs = pem.parse(chain_pem_bytes)
    if not parsed_certs:  # If not certs to parse, then raise error.
            raise ParseCertificateError(_UNABLE_TO_LOAD_CERTIFICATE)

    result = []
    for cert in parsed_certs:
        try:
            x509_cert = x509.load_pem_x509_certificate(
                cert.as_bytes(), default_backend()
            )
            result.append(x509_cert)
        except Exception:
            raise ParseCertificateError(_UNABLE_TO_LOAD_CERTIFICATE)

    return result # At this point at least one cert is parsed. There is no need to check result.
```

Reply:
that is correct and more clear, I made the change.